src/EditorFeatures/Core/Implementation/InlineRename/IEditorInlineRenameService.cs

```diff
@@ -165,6 +165,19 @@ internal interface IInlineRenameInfo
         /// </summary>
         Glyph Glyph { get; }
 
+        /// <summary>
+        /// The symbol kind of the symbol being renamed, for use in determining whether
+        /// the file can also be renamed
+        /// </summary>
+        SymbolKind SymbolKind { get; }
```

Comment:
What about typescript and F#? it looks like we set this to be its default value so renaming the document will never work for those languages.  It also can't work form XAML either

Reply:
I guess those services could pass the right symbol kind (they could keep a const of which enum is equivalent to typekind) but it seems easier to just have a method that the implementer provides to explain what is expected

Reply:
Maybe if I change to `FileRenameInfoEnum GetFileRenameInfo()` or similar so it's encapsulated as having complex enough logic that implementors can customize how they see fit. 

I like this plan, I'll start on it and see how it goes.